ros2_ws/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+cmake_minimum_required(VERSION 3.5)
+project(pointcloud2_pubsub)
+
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(sensor_msgs REQUIRED)
+
+add_executable(publisher_node src/publisher_node.cpp)
```

Comment:
Consider adding a line after `add_executable` to specify the include directories for the project. This will ensure the compiler can find necessary header files, especially if they're not directly in the `src` directory.  For example: `target_include_directories(publisher_node PUBLIC include)` (assuming your headers are in an `include` directory within your ROS2 workspace.)  Repeat this for `subscriber_node` on line 11.